storybookjs/storybook
# Docs: Fix rendering nested MDX

Closes #11542



## What I did

Traversing JSX tree when replacing mdx elements with built in react types should not be stopped, but it should visit all nodes. When there was MDXCreateElement inside custom component, element was not replaced and generated code snippet contained MDXCreateElement tags. Now it runs on all nodes and replaces all mdx elements.

## How to test

1. Write story with built in react element types are used inside custom component
2. Click on 'show code' to inspect snippet code
3. Snippet should not contain MDXCreateElement tags

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
